### DISCUSSION TITLE
Hello World Help

### DISCUSSION THREAD

**Discussion by laynr** [TorchIO-project/torchio#631] [Q&A] [answered]
Trying to follow along with https://torchio.readthedocs.io/quickstart.html#hello-world  and have a couple questions.

1.  First the most basic.  How do you train a model with the output of `inputs` and `target`?  I am coming from tensorflow and have been unable to figure this out even with the other examples in the project.  If you could point to another tutorial I would appreciate it.
2. It would be helpful if the example input could be provided.  Specifically 'subject_a.nii.gz', 'subject_a.nii', 'subject_b_dicom_folder', and 'subject_b_seg.nrrd' in a Colab. I understand nii.gz and a folder of dicoms, but less familiar with the format of the label input (.nill and .nrrd').  In my own project I have labels stored as a class 'torch.Tensor' for example tensor([0, 1, 1, 1]).  And when I call [tio.DATA]  I get an 'IndexError: too many indices for tensor of dimension 1'
3. For background I am working thru the RSNA Kaggle competition with this notebook https://www.kaggle.com/ohbewise/torchio?scriptVersionId=72629494 I  have no delusions of doing well, just here to learn. Ultimately want to do transfer learning with your highres3dnet model.

Thank you

**Comment by fepegar** [comment, marked as answer]
Hi, @laynr.

> First the most basic. How do you train a model with the output of inputs and target? I am coming from tensorflow and have been unable to figure this out even with the other examples in the project. If you could point to another tutorial I would appreciate it.

Sure! I suggest taking a look at the Tutorials section, right below the "Hello, World!" section you linked to.

> It would be helpful if the example input could be provided. Specifically 'subject_a.nii.gz', 'subject_a.nii', 'subject_b_dicom_folder', and 'subject_b_seg.nrrd' in a Colab. I understand nii.gz and a folder of dicoms, but less familiar with the format of the label input (.nill and .nrrd'). In my own project I have labels stored as a class 'torch.Tensor' for example tensor([0, 1, 1, 1]). And when I call [tio.DATA] I get an 'IndexError: too many indices for tensor of dimension 1'

Those filenames were chosen to show off the different formats that can be used (thanks to SimpleITK and NiBabel). There are loads of self-contained examples in the tutorials and the documentation.

The format `.nill` is not shown. If you mean `.nii`, it's just an uncompressed `.nii.gz`. `.nrrd` is just another format, more generic than NIfTI (`.nii`).

If you're getting an error somewhere, please report what you did, what you expected and what you got, with a minimally working example.
